crates/oxc_linter/src/lib.rs

```diff
@@ -466,8 +468,13 @@ impl Linter {
                     ));
                 }
             }
-            Err(_err) => {
-                // TODO: report diagnostic
+            Err(err) => {
+                let path = path.to_string_lossy();
+                let message = format!("Error running JS plugin.\nFile path: {path}\n{err}");
+                ctx_host.push_diagnostic(Message::new(
+                    OxcDiagnostic::error(message),
+                    PossibleFixes::None,
+                ));
```

Comment:
@camc314 Can I just check this is correct? Or can an `OxcDiagnostic::error` get filtered out in some cases?